src/Element.cc

```diff
@@ -93,6 +94,31 @@ void Element::SetCopyChildren(bool _value)
   this->dataPtr->copyChildren = _value;
 }
 
+/////////////////////////////////////////////////
+void Element::SetExplicitlySetInFile(const bool _value)
+{
+  this->dataPtr->explicitlySetInFile = _value;
+
+  ElementPtr element_ptr = this->GetFirstElement();
+  if (element_ptr != nullptr)
+  {
+    element_ptr->SetExplicitlySetInFile(_value);
+  }
+
+  element_ptr = this->GetNextElement();
```

Comment:
This looks like it updates only the first and second child elements. Does it work if the element has more than 2 children? Do you mind adding a test to verify?

Reply:
It's a recursive function so it should work for any depth and length of the element tree. I added one more level of depth to the test and siblings at each level: 52cd85f7c78a366d72b266a842c372c48f1588ce

Reply:
Ah, I mistakenly thought `GetNextElement` was getting the next child element. Instead, it gets the sibling. So the problem now is that `SetExplicitlySetInFile` will be called on all the siblings of the element the original element, which is not what we want. I have added some more expectations in the test that show the problem and created a PR on your fork https://github.com/marcoag/sdformat/pull/1. My suggestion would be to simply iterate on `this->dataPtr->elements` and call `SetExplicitlySetInFile`.

Reply:
Good point, this should be addressed at https://github.com/osrf/sdformat/pull/542/commits/369ea0dc89e94f9cd601cda84d707f17230d7cf1